### ISSUE TITLE
VfsClassLoaderTest error

### ISSUE THREAD

**Issue by jmark99** [apache/accumulo#3279]
**Describe the bug**

When attempting to perform some testing of the 1.10.3-RC0 candidate, I''m seeing a test failure in the VfsClassLoaderTest class.

**Versions (OS, Maven, Java, and others, as appropriate):**
 - Version 1.10.3-RC0
 - OS: Ubuntu 21.10
 - JDK: openjdk version "11" 2018-09-25; OpenJDK Runtime Environment 18.9 (build 11+28)

**To Reproduce**
1. Download the RC0 artifacts
2. `xzf accumulo-1.10.3-src.tar.gz`
3. `cd accumulo-1.10.3`
4. `mvn clean package`
5. Saw the following output:
<pre>
[INFO] Running org.apache.accumulo.start.classloader.vfs.providers.VfsClassLoaderTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.673 s <<< FAILURE! - in org.apache.accumulo.start.classloader.vfs.providers.VfsClassLoaderTest
[ERROR] org.apache.accumulo.start.classloader.vfs.providers.VfsClassLoaderTest  Time elapsed: 1.653 s  <<< ERROR!                                                       
java.lang.ExceptionInInitializerError                                                                                                                                   
Caused by: java.lang.StringIndexOutOfBoundsException: begin 0, end 3, length 2             
</pre>

**Additional context**

Looking into the surefire-reports directory similar information was provided:

<pre>
The output in the surefire-reports directory doesn't provide too much info.

Test set: org.apache.accumulo.start.classloader.vfs.providers.VfsClassLoaderTest
-------------------------------------------------------------------------------
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.716 s <<< FAILURE! - in org.apache.accumulo.start.classloader.vfs.providers.VfsClassLoaderTest
org.apache.accumulo.start.classloader.vfs.providers.VfsClassLoaderTest  Time elapsed: 1.698 s  <<< ERROR!
java.lang.ExceptionInInitializerError
Caused by: java.lang.StringIndexOutOfBoundsException: begin 0, end 3, length 2

[main} INFO  org.apache.hadoop.hdfs.MiniDFSCluster  - starting cluster: numNameNodes=1, numDataNodes=1
[main} INFO  org.apache.hadoop.hdfs.MiniDFSCluster  - Shutting down the Mini HDFS Cluster
</pre>

In order to find additional information, I attempted to install Accumulo using fluo-uno.

<pre>
$ uno fetch accumulo
hadoop-2.6.5.tar.gz download matches expected checksum (fd9d24155bb18475c25bc6a88284a696224d5ef38503c5c3f1d4c0b65ae1c552d90d2136ca0fa45294d62204eed3d1154785c12cc260cca4
2ea2fbf027b461b2)                                                                                                                                                       
zookeeper-3.4.14.tar.gz download matches expected checksum (b2e03d95f8cf18b97a46e2f53871cef5a5da9d5d80b97009375aed7fb35368c440ca944c7e8b64efabbc065f6fb98bb86239f7c1491f
0490efc71876d5a7f424)                                                                                                                                                   
accumulo-1.10.3-bin.tar.gz download matches expected checksum (6a57133cdc8fe55076de9be2deb6faf0d170c86f)      
$ uno setup accumulo                                                                                                                 
Setting up accumulo (detailed logs in /home/user/dev/fluo-uno/install/logs/setup)...                                                                                    
Installing Apache Hadoop 2.6.5 at /home/user/dev/fluo-uno/install/hadoop-2.6.5                                                                                          
Installing Apache ZooKeeper 3.4.14 at /home/user/dev/fluo-uno/install/zookeeper-3.4.14                                                                                  
Installing Apache Accumulo 1.10.3 at /home/user/dev/fluo-uno/install/accumulo-1.10.3                                                                                    
Apache Accumulo 1.10.3 is running                                                                                                                                       
    * Accumulo Monitor: http://localhost:9995/                                                                                                                          
    * view logs at /home/user/dev/fluo-uno/install/logs/accumulo                                                                                                        
Setup of accumulo complete.                                                              
$ uno status                                                                                                                         
Accumulo processes running: master(1873416) gc(1873488) tracer(1873560)                                                                                                 
Hadoop processes running: localhost(1873416) localhost(1873488) localhost(1873560) 
</pre>

ZooKeeper failed to start. So I ran `uno start accumulo` to see if I could capture more info.

<pre>
$ uno start accumulo
ZooKeeper JMX enabled by default
Using config: /home/user/dev/fluo-uno/install/zookeeper-3.4.14/bin/../conf/zoo.cfg
Starting zookeeper ... STARTED
Exception in thread "main" java.lang.ExceptionInInitializerError
        at org.apache.hadoop.util.GenericOptionsParser.preProcessForWindows(GenericOptionsParser.java:438)
        at org.apache.hadoop.util.GenericOptionsParser.parseGeneralOptions(GenericOptionsParser.java:484)
        at org.apache.hadoop.util.GenericOptionsParser.<init>(GenericOptionsParser.java:170)
        at org.apache.hadoop.util.GenericOptionsParser.<init>(GenericOptionsParser.java:153)
        at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:64)
        at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:84)
        at org.apache.hadoop.hdfs.tools.GetConf.main(GetConf.java:331)
Caused by: java.lang.StringIndexOutOfBoundsException: begin 0, end 3, length 2
        at java.base/java.lang.String.checkBoundsBeginEnd(String.java:3319)
        at java.base/java.lang.String.substring(String.java:1874)
        at org.apache.hadoop.util.Shell.<clinit>(Shell.java:50)
        ... 7 more
Starting namenodes on []
localhost: starting namenode, logging to /home/user/dev/fluo-uno/install/logs/hadoop/hadoop-user-namenode-devlap.out
localhost: starting datanode, logging to /home/user/dev/fluo-uno/install/logs/hadoop/hadoop-user-datanode-devlap.out
Exception in thread "main" java.lang.ExceptionInInitializerError
        at org.apache.hadoop.util.GenericOptionsParser.preProcessForWindows(GenericOptionsParser.java:438)
        at org.apache.hadoop.util.GenericOptionsParser.parseGeneralOptions(GenericOptionsParser.java:484)
        at org.apache.hadoop.util.GenericOptionsParser.<init>(GenericOptionsParser.java:170)
        at org.apache.hadoop.util.GenericOptionsParser.<init>(GenericOptionsParser.java:153)
        at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:64)
        at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:84)
        at org.apache.hadoop.hdfs.tools.GetConf.main(GetConf.java:331)
Caused by: java.lang.StringIndexOutOfBoundsException: begin 0, end 3, length 2
        at java.base/java.lang.String.checkBoundsBeginEnd(String.java:3319)
        at java.base/java.lang.String.substring(String.java:1874)
        at org.apache.hadoop.util.Shell.<clinit>(Shell.java:50)
        ... 7 more
starting yarn daemons
starting resourcemanager, logging to /home/user/dev/fluo-uno/install/logs/hadoop/yarn-user-resourcemanager-devlap.out
localhost: starting nodemanager, logging to /home/user/dev/fluo-uno/install/logs/hadoop/yarn-user-nodemanager-devlap.out
Starting monitor on localhost
Starting tablet servers .... done
OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
Starting tserver on localhost
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.hadoop.security.authentication.util.KerberosUtil (file:/home/user/dev/fluo-uno/install/hadoop-2.6.5/share/hadoop/common
/lib/hadoop-auth-2.6.5.jar) to method sun.security.krb5.Config.getInstance()
WARNING: Please consider reporting this to the maintainers of org.apache.hadoop.security.authentication.util.KerberosUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
2023-04-06 15:21:02,511 [start.Main] ERROR: Thread 'org.apache.accumulo.master.state.SetGoalState' died.
java.lang.ExceptionInInitializerError
        at org.apache.hadoop.util.StringUtils.<clinit>(StringUtils.java:79)
        at org.apache.hadoop.security.Groups.parseStaticMapping(Groups.java:116)
        at org.apache.hadoop.security.Groups.<init>(Groups.java:93)
        at org.apache.hadoop.security.Groups.<init>(Groups.java:73)
        at org.apache.hadoop.security.Groups.getUserToGroupsMappingService(Groups.java:293)
        at org.apache.hadoop.security.UserGroupInformation.initialize(UserGroupInformation.java:283)
        at org.apache.hadoop.security.UserGroupInformation.ensureInitialized(UserGroupInformation.java:260)
        at org.apache.hadoop.security.UserGroupInformation.loginUserFromSubject(UserGroupInformation.java:789)
        at org.apache.hadoop.security.UserGroupInformation.getLoginUser(UserGroupInformation.java:774)
        at org.apache.hadoop.security.UserGroupInformation.getCurrentUser(UserGroupInformation.java:647)
        at org.apache.hadoop.fs.FileSystem$Cache$Key.<init>(FileSystem.java:2755)
        at org.apache.hadoop.fs.FileSystem$Cache$Key.<init>(FileSystem.java:2747)
        at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:2613)
        at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:370)
        at org.apache.hadoop.fs.Path.getFileSystem(Path.java:296)
        at org.apache.accumulo.core.volume.VolumeImpl.<init>(VolumeImpl.java:45)
        at org.apache.accumulo.core.volume.VolumeConfiguration.create(VolumeConfiguration.java:161)
        at org.apache.accumulo.server.fs.VolumeManagerImpl.get(VolumeManagerImpl.java:357)
        at org.apache.accumulo.server.fs.VolumeManagerImpl.get(VolumeManagerImpl.java:339)
        at org.apache.accumulo.server.fs.VolumeManagerImpl.get(VolumeManagerImpl.java:333)
        at org.apache.accumulo.master.state.SetGoalState.main(SetGoalState.java:46)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.apache.accumulo.start.Main$2.run(Main.java:170)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.StringIndexOutOfBoundsException: begin 0, end 3, length 2
        at java.base/java.lang.String.checkBoundsBeginEnd(String.java:3319)
        at java.base/java.lang.String.substring(String.java:1874)
        at org.apache.hadoop.util.Shell.<clinit>(Shell.java:50)
        ... 27 more
Starting master on localhost
Starting gc on localhost
Starting tracer on localhost
</pre>

**Comment by ddanielr** [comment]
@jmark99 I downloaded the same candidate file and ran through your steps but I'm not able to reproduce your errors.

I am on a different OS and java version. 
```
$ cat /etc/redhat-release 
CentOS Linux release 7.9.2009 (Core)

$ java -version
openjdk version "1.8.0_352"
OpenJDK Runtime Environment (build 1.8.0_352-b08)
OpenJDK 64-Bit Server VM (build 25.352-b08, mixed mode)
```

Test output from  the accumulo-start section of `mvn clean package` 
```
[INFO] --- exec-maven-plugin:3.0.0:exec (Build Test jars) @ accumulo-start ---
[INFO] 
[INFO] --- exec-maven-plugin:3.0.0:exec (Build HelloWorld jars) @ accumulo-start ---
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M4:test (default-test) @ accumulo-start ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.accumulo.start.classloader.vfs.providers.VfsClassLoaderTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 25.488 s - in org.apache.accumulo.start.classloader.vfs.providers.VfsClassLoaderTest
[INFO] Running org.apache.accumulo.start.classloader.vfs.AccumuloVFSClassLoaderTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.88 s - in org.apache.accumulo.start.classloader.vfs.AccumuloVFSClassLoaderTest
[INFO] Running org.apache.accumulo.start.classloader.vfs.AccumuloReloadingVFSClassLoaderTest
[WARNING] Tests run: 4, Failures: 0, Errors: 0, Skipped: 2, Time elapsed: 14.489 s - in org.apache.accumulo.start.classloader.vfs.AccumuloReloadingVFSClassLoaderTest
[INFO] Running org.apache.accumulo.start.classloader.vfs.ContextManagerTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.511 s - in org.apache.accumulo.start.classloader.vfs.ContextManagerTest
[INFO] Running org.apache.accumulo.start.MainTest
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.302 s - in org.apache.accumulo.start.MainTest
[INFO] 
[INFO] Results:
[INFO] 
[WARNING] Tests run: 18, Failures: 0, Errors: 0, Skipped: 2
```

**Comment by EdColeman** [comment]
Running the test individually and `mvn clean package` on the downloaded src file both pass for me.

```
mvn --version
Apache Maven 3.6.3
Maven home: /usr/share/maven
Java version: 11.0.18, vendor: Eclipse Adoptium, runtime: /usr/java/adopt-jdk-11/jdk-11.0.18+10
Default locale: en, platform encoding: UTF-8
OS name: "linux", version: "5.19.0-1022-aws", arch: "amd64", family: "unix"

```

**Comment by ctubbsii** [comment]
I also haven't reproduced this issue. Please make sure your Uno is up-to-date, as that seems the most likely problem. If not, then please check to see if #3280 fixes your issue.

**Comment by cshannon** [comment]
Just to add one more person, the test also passes for me as well when testing on RHEL 8 and JDK 11.

**Comment by jmark99** [comment]
I bumped the Jetty version, but still received the same error. But then I bumped the Hadoop version to a 2.7.7 and the error went away (regardless of Jetty version). Using Hadoop version 3.0.3 worked as well. The error seems to be confined to Hadoop 2.6.5 on my local system. Given the success of others running the tests, I'm okay closing the ticket given that it appears to be confined to some specific setting on my local system.
